docs/userguide/src/portingguide/perf_tuning/alloc.md

```diff
@@ -47,37 +47,55 @@ have an assertion to ensure that the native type has the exact same layout as th
+
 Note that the `allocate_default` closure in the example below assumes the allocation semantics is `AllocationSemantics::Default`
 and its selected allocator uses bump-pointer allocation.
 Real-world fast-path implementations for high-performance VMs are usually JIT-compiled, inlined, and specialized for the current plan
-and allocation site, so that the allocation semantics of the concrete allocation site (and therefore the selected allocator) is known to the JIT compiler.
+and allocation site. Hence, the allocation semantics of the concrete allocation site (and therefore the selected allocator) is known to the JIT compiler.
 
 For the sake of simplicity, we only store _one_ `BumpPointer` in the TLS in the example.
 In MMTk, each plan has multiple allocators, and the allocation semantics are mapped
-to those allocator by the GC plan you choose. So a plan use multiple allocators, and
+to those allocator by the GC plan you choose. So a plan uses multiple allocators, and
 depending on how many allocation semantics are used by a binding, the binding may use multiple allocators as well.
-In practice, a binding may embed multiple fastpath structs as the example for those allocators if they would like
+In practice, a binding may embed multiple fast-path structs for all the allocators they use if they would like
 more efficient allocation.
 
-Also for simpliticy, the example assumes the default allocator for the plan in use is a bump pointer allocator.
+Also for simplicity, the example assumes the default allocator for the plan in use is a bump pointer allocator.
 Many plans in MMTk use bump pointer allocator for their default allocation semantics (`AllocationSemantics::Default`),
 which includes (but not limited to) `NoGC`, `SemiSpace`, `Immix`, generational plans, etc.
-If a plan does not do bump-pointer allocation, we may still implement fast paths, but we need to embed different data structures instead of `BumpPointer`.
+If a plan does not do bump-pointer allocation, we may still implement fast-paths, but we need to embed different data structures instead of `BumpPointer`.
+
+```rust
+{{#include ../../../../../vmbindings/dummyvm/src/tests/doc_mutator_storage.rs:mutator_storage_embed_fast-path_struct}}
+```
 
+And pseudo-code for how you would reset the `BumpPointer`s for all mutators in `resume_mutators`. Note that these mutators are the runtime's actual mutator threads (i.e. where the cached bump pointers are stored) and are different from MMTk's `Mutator` struct.
 ```rust
-{{#include ../../../../../vmbindings/dummyvm/src/tests/doc_mutator_storage.rs:mutator_storage_embed_fastpath_struct}}
+impl Collection<RtName> for RtNameCollection {
+  ...
+  fn resume_mutators(tls: VMWorkerThread) {
+    // Reset the cached bump pointers of each mutator to 0x0 after a GC to
+    // ensure that the VM sees a cohesive state
```

Comment:
A BumpPointer has a cursor and a limit.  It is not clear what 'reset bump pointer to 0' means.

```suggestion
    // Reset the cached bump pointers of each mutator (setting both cursor and limit to 0) after a GC to
    // ensure that the VM sees a cohesive state
```